README
Pyrename version 0.8 2012-08-30
AUTHOR: Gianluca Pernigotto
LICENCE: Copyright (c) 2012 Gianluca Pernigotto GPL3 <jeanlucperni@gmail.com>



DESCRIZIONE: 

   Pyrename è scritto completamente in python, si usa a riga di comando e fra le
   opzioni è disponibile anche la numerazione progressiva.  Ogni processo di 
   rinominazione viene applicato separatamente per files e cartelle. 
   

DIPENDENZE:

   python=2.6 >
  

FEATURES:
 
tipi di rinomina con numerazione progressiva:

    - Rinomina con un nome e assegna automaticamente l'ordine numerico.

    - Aggiunge ai nomi di origine una numerazione progressiva mantenendo 
      l'ordine alfanumerico. I nomi di origine e le estensioni non vengono 
      modificati.

    - Rimuove la numerazione progressiva precedentemente applicata con i
      metodi sopra.

tipi di rinomina con modifiche a stringhe:

    - Cambia o aggiunge determinate stringhe ai nomi con il medesimo match. 
      Con l'opzione è possibile modificare anche l'estensione dei file, sebbene 
      sia incoraggiato usare il metodo appropriato (addext o extreplace) ed
      è inoltre possibile eliminarle.

    - Aggiunge delle stringhe ai nomi senza nessun tipo di match (i nomi di 
      origine possono essere diversi per ogni file o cartella), consentendoci 
      di decidere anche la posizione della nuova stringa, cioè all'inizio o 
      alla fine del nome originale (però non dell'estensione che invece 
      rimane uguale). 

tipi di rinomina alle estensioni: 

    - Sostituisce le estensioni incontrate in un match: la rinomina delle 
      estensioni avviene immettendo l'estensione che si vuol cambiare in un 
      gruppo di file/cartelle.

    - Aggiunge le estensioni ai file/cartelle che ne sono privi. Specificando 
      il prefisso ('-f' o '-d') se ne determina il risultato per file o cartelle.



CHANGELOG
+------------------------------------+
December 12 2017  Version 0.9.1
+------------------------------------+

- make a four spaces indentetion for all python scripts 
  and following the general PEP8
  
- update readme.md



+------------------------------------+
January 28 2015  Version 0.9
+------------------------------------+

- This is first stable release

- Translation done completely in English 

- Make a setup.py and MANIFEST.in for building a package for your Linux 
  distribution (Debian and derived, Slackware)


+------------------------------------+
August 30 2012  Version 0.8
+------------------------------------+
- rename: Risolto bug nella risoluzione dei nomi esistenti.
          Ora il controllo è completo e più sicuro.

- Aggiunto il metodo 'delnumerate' per la rimozione numerica progressiva.

- Aggiornamento del manuale 'man pyrename'

- Aggiunta di qualche elemento cosmetico


+------------------------------------+
August 29 2012  Version 0.7
+------------------------------------+
- extreplace: Eliminata funzione che rinominava e aggiungeva l'estensione a tutti i file/dir

- extreplace, addext: Risolto grave bug che sovrascriveva file/dir con lo stesso nome.
		      Un consistente controllo viene eseguito prima della rinomina.

- extreplace, addext: Risolto bug nel campo di input per la rinomina delle estens. 
		      Ora l'input non può più essere vuoto.

- extreplace: Risolto bug per cui l'aggiunta di estensione rinominava file/dir con lo stesso nome dell'estensione.
	      Ora i file/dir senza estensione vengono esclusi.

- Aggiornamento del manuale 'man pyrename'


+------------------------------------+
August 23 2012  Version 0.6
+------------------------------------+
- Eliminata funzione obsoleta 'numreplace'

- Aggiunto nelle funzioni di numerazione 'namereplace' ed 'enumerate' l'opzione 'tail' e 'head'

- Aggiornamento del manuale 'man pyrename' 


+------------------------------------+
August 19 2012  Version 0.5
+------------------------------------+
- Risolto grave bug nelle funzioni 'rename' ed 'addname' che sovrascrivevano file o cartelle con lo stesso nome.

- Aggiunto metodi 'addext' ed 'ext_replace' per la rinominazione delle estensioni

- Aggiornamento del manuale 'man pyrename' 

- Cambiata la struttura del package, ora il binario riceve solo argomenti mentre le funzioni sono nei moduli.

- Rimossi file per 'help'


+------------------------------------+
Sept. 21  2010  Version 0.4
+------------------------------------+

Initial release
 


renamerpack/file_renamer.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
##############################################
# PROGRAM-NAME: pyrename
# VERSION:      v0.9
# DESCRIPTION:  module batch renaming for files
# AUTHOR:       Gianluca jeanslack Pernigotto <jeanlucperni@gmail.com>
# LICENSE:      Copyright (c) Gianluca jeanslack Pernigotto GPL 3 
# FIRST RELEASE: v0.8 (date 30/08/2012)
# LAST REVISION:  28/01/2015/12/12/2017
##############################################
import os
import sys
import string
######################
def f_namereplace():
    """
    rinomina e numera tutti i file trovati
    """
    FileList = []
    dirname = os.getcwd()  
    for name in os.listdir(dirname):
        if os.path.isfile(os.path.join(dirname, name)): # filtra il contenuto
            FileList.append(name)
    FileList.sort()
    dest = raw_input("\033[1mRename all folders as: >\033[0m ")
    disp = raw_input("\033[0;1mChoose where to place the numbering > "
                    "[head/tail] \033[0m ")
    if disp == 'head' or disp == 'HEAD':
        for i, filename in enumerate(FileList):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, str(i+1)+'_'+dest+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more folders have "
                        "the same name\033[0m")
        sys.exit() 

    elif disp == 'tail' or disp == 'TAIL':
        for i, filename in enumerate(FileList):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, dest+'_'+str(i+1)+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more folders have "
                        "the same name\033[0m")
        sys.exit()
    else:
        sys.exit('\033[31;1m[ERROR]: bad input (tail/head)\033[0m')

    sys.exit()

######################
def f_rename():
    """
    rinomina qualsiasi stringa di un file in un match
    """
    src = raw_input("\033[1m Enter the string to be replaced >\033[0m ")
    dest = raw_input("\033[1m Insert the replacement string >\033[0m ")
    li = os.listdir(os.getcwd())
    FileList = []

    for name in li:		
        if os.path.isfile(os.path.join(name)): # filtra il contenuto
            nuovoNome = name.replace("%s"%(src),"%s"%(dest))
            FileList.append(nuovoNome)

    for i in FileList: # itero sull'esistenza dei file
        if FileList.count(i) > 1:
            sys.exit("\033[31;1m[ERROR]:  There is a file with the same "
                    "name\033[0m")

    if src in li and os.path.isdir(src):
        print ("\033[31;1m[WARNING]: I can not rename the directory, use "
            "prefix -d\033[0m")
        sys.exit()		
    else:
        for filename in li:
            if os.path.isfile(filename):
                nuovoNome = filename.replace("%s"%(src),"%s"%(dest))
                try:
                    os.rename(filename, nuovoNome)
                except OSError: #'[Errno 21] Is a directory'
                    sys.exit("\033[31;1m[ERROR]: There is a directory "
                            "with the same name\033[0m")
    sys.exit()

######################
def f_enumerate():
    """
    numera i file e conserva il nome ed estensione
    """
    FileList = []
    dirname = os.getcwd()
    for name in os.listdir(dirname):
        if os.path.isfile(os.path.join(dirname, name)): # filtra il contenuto
            FileList.append(name)
    FileList.sort()
    disp = raw_input("\033[0;1mChoose where to place the numbering > "
                    "[head/tail] \033[0m ")
    if disp == 'head' or disp == 'HEAD':
        for i, filename in enumerate(FileList):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, str(i+1)+'_'+name+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more directories have "
                        "the same name\033[0m")
        sys.exit() 
    elif disp == 'tail' or disp == 'TAIL':
        for i, filename in enumerate(FileList):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, name+'_'+str(i+1)+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more directories have "
                        "the same name\033[0m")
        sys.exit()
    else:
        sys.exit('\033[31;1m[ERROR]: bad input\033[0m')

    sys.exit()

#####################
def f_delnumerate():
    """
    toglie la numerazione progressiva
    """
    li = os.listdir(os.getcwd())
    FileList = []

    for name in li:
        if os.path.isfile(os.path.join(name)): # filtra il contenuto
            FileList.append(name)
    FileList.sort()

    tmp = []
    for i, name in enumerate(FileList):
        nuovoNome = name.replace(str(i+1),"")
        tmp.append(nuovoNome)

    for i in tmp:
        if tmp.count(i) > 1:
            sys.exit("\033[31;1m[ERROR]: One or more files have the same "
                    "name\033[0m")

    for i, filename in enumerate(FileList):
        #name, ext = os.path.splitext(filename)
        nuovoNome = filename.replace(str(i+1),"")
        try:
            os.rename(filename, nuovoNome)
        except OSError: #'[Errno 21] Is a directory'
            sys.exit("\033[31;1m[ERROR]: One or more directories have the "
                    "same name\033[0m")
    sys.exit()

#######################
def f_addname():
    """
    Aggiunge stringhe all'inizio o alla fine dei nomi preservando 
    l'estensione
    """
    dest = raw_input("\033[0;1mEnter the string to be added to the "
                    "file name >\033[0m ")
    FileList = []
    dirname = os.getcwd()  
    for name in os.listdir(dirname):
        if os.path.isfile(os.path.join(dirname, name)): 
            FileList.append(name)
    FileList.sort()

    disp = raw_input("\033[0;1mChoose where to place the new string > "
                    "[head/tail] \033[0m ")
    if disp == 'head' or disp == 'HEAD':
        for a in FileList:
            if "%s%s" % (dest, a) in FileList:
                sys.exit("\033[31;1m[ERROR]: The object file already "
                        "exists with the same name \033[0m") 
        for i, filename in enumerate(FileList):
            try:
                os.rename(filename, dest+filename)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: There is a directory with "
                        "the same name\033[0m")

    elif disp == 'tail' or disp == 'TAIL':
        for a in FileList:
            if "%s%s" % (a, dest) in FileList:
                sys.exit("\033[31;1m[ERROR]: The object file already "
                        "exists with the same name\033[0m")
        for i, filename in enumerate(FileList):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, name+dest+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: There is a directory with "
                        "the same name\033[0m")
    else:
        sys.exit('\033[31;1m[ERROR]: bad input\033[0m')
    sys.exit()

#####################
def f_extreplace():
    """
    rinomina l'estensione dei file in un match
    """
    oldext = raw_input("\033[1mEnter the extension to change>\033[0m .")
    newext = raw_input("\033[1mEnter the new extension>\033[0m .")
    FileExtension = []
    FileExist = []
    li = os.listdir(os.getcwd())

    if oldext in string.whitespace:
        sys.exit("\033[31;1m[ERROR]: Empty string (old ext) \033[0m")
    if newext in string.whitespace:
        sys.exit("\033[31;1m[ERROR]: Empty string (new ext) \033[0m")

    for filename in li: # appendo solo le ext. dei file
        if os.path.isfile(os.path.join(filename)):
            name, ext = os.path.splitext(filename)
            FileExtension.append(ext)
    for filename in li: # appendo tutti i file/dir esistenti con la nuova ext. 
        os.path.join(filename)
        name, ext = os.path.splitext(filename)
        if ext.endswith(oldext):
            FileExist.append("%s.%s" %(name,newext))
    # itero sull'esistenza di file con uguale nome.newext
    for filename in FileExist: 
        if os.path.exists(filename):
            print ("\033[31;1m[ERROR]: The object already exists with the "
                "same name\033[0m")
            if os.path.isfile(filename):
                sys.exit('        \033[31;1m  and is a file\033[0m')
            elif os.path.isdir(filename):
                sys.exit('        \033[31;1m  and is a directory\033[0m')
    if ".%s" % oldext not in FileExtension:
        sys.exit("\033[31;1m[ERROR]: There is no file with the "
                "extension '.%s' in the path\033[0m" % oldext)
    for filename in li:
        if os.path.isfile(os.path.join(filename)):
            name, ext = os.path.splitext(filename)
            if ext.endswith(oldext):			
                newname = filename.replace(oldext,newext)
                try:
                    os.rename(filename, newname)
                except OSError: #'[Errno 21] Is a directory'
                    sys.exit("\033[31;1m[ERROR]: There is a directory "
                            "with the same name\033[0m")
    sys.exit()

#####################
def f_addext():
    """
    aggiunge l'estensione ai file che ne sono privi
    """
    newext = raw_input("\033[1m Enter the extension to apply>\033[0m .")
    FileList = []
    FileExist = []
    li = os.listdir(os.getcwd())
    if newext in string.whitespace:
        sys.exit("\033[31;1m[ERROR]: Empty string (new ext) \033[0m")

    for filename in li:
        if os.path.isfile(os.path.join(filename)):
            name, ext = os.path.splitext(filename)
            if ext in string.whitespace:
                FileList.append(filename)	
    if FileList == []:
        sys.exit("\033[31;1m[ERROR]: There is no file with no extension "
                "in the path\033[0m")

    for filename in li:
        if os.path.join(filename):
            name, ext = os.path.splitext(filename)
            FileExist.append("%s.%s" %(name,newext))
    print FileExist
    for verify in FileExist:	
        if FileExist in li:
            print ("\033[31;1m[ERROR]: The object already exists with the "
                "same name\033[0m")
            if os.path.isfile(verify):
                sys.exit('        \033[31;1m  and is a file\033[0m')
            elif os.path.isdir(verify):
                sys.exit('        \033[31;1m  and is a directory\033[0m')
    for filename in FileList:
        try:
            os.rename(filename, "%s.%s" % (filename, newext))	
        except OSError: #'[Errno 21] Is a directory'
            sys.exit("\033[31;1m[ERROR]: There is a directory with the "
                    "same name\033[0m")
    sys.exit()


renamerpack/dir_renamer.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
#############################################
# PROGRAM-NAME: pyrename
# VERSION: v0.9
# DESCRIPTION: module batch renaming for folders
# AUTHOR: Gianluca jeanslack Pernigotto <jeanlucperni@gmail.com>
# LICENSE: Copyright (c) Gianluca jeanslack Pernigotto GPL 3 
# FIRST RELEASE: v0.8 (date 30/08/2012)
# LAST REVISION:  28/01/2015/12/12/2017
##############################################
import os
import sys
import string

###################### 
def d_namereplace():
    """
    rinomina e numera tutte le cartelle trovate
    """
    filelist = []
    dirname = os.getcwd()  
    for name in os.listdir(dirname):
        if os.path.isdir(os.path.join(dirname, name)): # filtra il contenuto
            filelist.append(name)
    filelist.sort()
    dest = raw_input("\033[1mRename all folders as: >\033[0m ")
    disp = raw_input("\033[0;1mChoose where to place the numbering > "
                     "[head/tail] \033[0m ")
    if disp == 'head' or disp == 'HEAD':
        for i, filename in enumerate(filelist):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, str(i+1)+'_'+dest+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more files have the "
                         "same name\033[0m")
        sys.exit() 
    elif disp == 'tail' or disp == 'TAIL':
        for i, filename in enumerate(filelist):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, dest+'_'+str(i+1)+ext) 
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more files have the "
                         "same name\033[0m")
        sys.exit()
    else:
        sys.exit('\033[31;1m[ERROR]: bad input\033[0m')
        
    sys.exit()

######################
def d_rename():
    """
    rinomina qualsiasi stringa di un file in un match
    """
    src = raw_input("\033[1m Enter the string to be replaced >\033[0m ")
    dest = raw_input("\033[1m Insert the replacement string >\033[0m ")
    li = os.listdir(os.getcwd())
    FileList = []
    
    for name in li:		
        if os.path.isdir(os.path.join(name)): # filtra il contenuto
            nuovoNome = name.replace("%s"%(src),"%s"%(dest))
            FileList.append(nuovoNome)
    
    for i in FileList: # itero sull'esistenza dei file
        if FileList.count(i) > 1:
            sys.exit("\033[31;1m[ERROR]: There is a directory with the "
                     "same name\033[0m")
        
    if src in li and os.path.isfile(src):
        print ("\033[31;1m[WARNING]: I can not rename the files, use the "
               "-f prefix\033[0m")
        sys.exit()
    else:
        for filename in li:
            if os.path.isdir(filename):
                nuovoNome = filename.replace("%s"%(src),"%s"%(dest))
                try:
                    os.rename(filename, nuovoNome)
                except OSError: #'[Errno 21] Is a directory'
                    sys.exit("\033[31;1m[ERROR]:  There is a file with the "
                             "same name\033[0m")
    sys.exit()

######################
def d_enumerate():
    """
    numera le cartelle e conserva il nome ed estensione
    """
    filelist = []
    dirname = os.getcwd()
    for name in os.listdir(dirname):
        if os.path.isdir(os.path.join(dirname, name)): # filtra il contenuto
            filelist.append(name)	
    filelist.sort()
    disp = raw_input("\033[0;1mChoose where to place the numbering > "
                     "[head/tail] \033[0m ")
    if disp == 'head' or disp == 'HEAD':
        for i, filename in enumerate(filelist):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, str(i+1)+'_'+name+ext) 
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more files have the "
                         "same name\033[0m")
        sys.exit() 
    elif disp == 'tail' or disp == 'TAIL':
        for i, filename in enumerate(filelist):
            name, ext = os.path.splitext(filename)
            try:
                os.rename(filename, name+'_'+str(i+1)+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]: One or more files have the "
                         "same name\033[0m")
        sys.exit()
    else:
        sys.exit('\033[31;1m[ERROR]: bad input\033[0m')

#####################
def d_delnumerate():
    """
    toglie la numerazione progressiva
    """
    li = os.listdir(os.getcwd())
    FileList = []
    
    for name in li:
        if os.path.isdir(os.path.join(name)): # filtra il contenuto
            FileList.append(name)
    FileList.sort()
    
    tmp = []			
    for i, name in enumerate(FileList):
        nuovoNome = name.replace(str(i+1),"")
        tmp.append(nuovoNome)

    for i in tmp: # itero sull'esistenza dei file
        if tmp.count(i) > 1:
            sys.exit("\033[31;1m[ERROR]: One or more folders have the "
                     "same name\033[0m")
    
    for i, filename in enumerate(FileList):
        #if os.path.isdir(os.path.join(filename)):
        nuovoNome = filename.replace(str(i+1),"")
        try:
            os.rename(filename, nuovoNome)
        except OSError: #'[Errno 21] Is a directory'
            sys.exit("\033[31;1m[ERROR]: One or more files have the "
                     "same name\033[0m")
    sys.exit()

#########
    
def f_addname():
    """
    Aggiunge stringhe all'inizio o alla fine dei nomi 
    preservando l'estensione
    """
    dest = raw_input("\033[0;1mEnter the string to be added to the "
                     "file name >\033[0m ")
    FileList = []
    dirname = os.getcwd()  
    for name in os.listdir(dirname):
        if os.path.isdir(os.path.join(dirname, name)): 
            FileList.append(name)
    FileList.sort()

    disp = raw_input("\033[0;1mChoose where to place the new string > "
                     "[head/tail] \033[0m ")
    if disp == 'head' or disp == 'HEAD':
        for a in FileList:
            if "%s%s" % (dest, a) in FileList:
                sys.exit("\033[31;1m[ERROR]: The object folder already "
                         "exists with the same name \033[0m") 
        for i, filename in enumerate(FileList):
            try:
                os.rename(filename, dest+filename)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]:  There is a file with the "
                         "same name\033[0m")

    elif disp == 'tail' or disp == 'TAIL':
        for a in FileList:
            if "%s%s" % (a, dest) in FileList:
                sys.exit("\033[31;1m[ERROR]: The object folder already "
                         "exists with the same name\033[0m")
        for i, filename in enumerate(FileList):
            name, ext = os.path.splitext(filename)			
            try:
                os.rename(filename, name+dest+ext)
            except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]:  There is a file with the "
                         "same name\033[0m")
    else:
        sys.exit('\033[31;1m[ERROR]: bad input\033[0m')
    sys.exit()
        
        
######################
    
def d_extreplace():
    """
    rinomina l'estensione delle cartelle in un match
    """
    oldext = raw_input("\033[1mEnter the extension to change>\033[0m .")
    newext = raw_input("\033[1mEnter the new extension>\033[0m .")
    FileExtension = []
    FileExist = []
    li = os.listdir(os.getcwd())
            
    if oldext in string.whitespace:
        sys.exit("\033[31;1m[ERROR]: Empty string (old ext) \033[0m")
    if newext in string.whitespace:
        sys.exit("\033[31;1m[ERROR]: Empty string (new ext) \033[0m")
                
    for filename in li: # appendo solo le ext. delle cartelle
        if os.path.isdir(os.path.join(filename)):
            name, ext = os.path.splitext(filename)
            FileExtension.append(ext)
    for filename in li: # appendo tutti i file/dir esistenti con la nuova ext.
        os.path.join(filename)
        name, ext = os.path.splitext(filename)
        if ext.endswith(oldext):
            FileExist.append("%s.%s" %(name,newext))
    # itero sull'esistenza di file con uguale nome.newext
    for filename in FileExist: 
        if os.path.exists(filename):
            print ("\033[31;1m[ERROR]: The object already exists with the "
                   "same name\033[0m")
            if os.path.isfile(filename):
                sys.exit('        \033[31;1m  and is a file\033[0m')
            elif os.path.isdir(filename):
                sys.exit('        \033[31;1m  and is a directory\033[0m')
        
    if ".%s" % oldext not in FileExtension:
        sys.exit("\033[31;1m[ERROR]: There is no file with the extension "
                 "'.%s' in the path\033[0m" % oldext)
        
    for filename in li:
        if os.path.isdir(os.path.join(filename)):
            name, ext = os.path.splitext(filename)
            if ext.endswith(oldext):			
                newname = filename.replace(oldext,newext)
                try:
                    os.rename(filename, newname)
                except OSError: #'[Errno 21] Is a directory'
                    sys.exit("\033[31;1m[ERROR]:  There is a file with the "
                             "same name\033[0m")
                
    sys.exit()

#####################
def d_addext():
    """
    aggiunge l'estensione alle cartelle che ne sono prive
    """
    newext = raw_input("\033[1m Enter the extension to apply>\033[0m .")
    FileList = []
    FileExist = []
    li = os.listdir(os.getcwd())
    if newext in string.whitespace:
        sys.exit("\033[31;1m[ERROR]: Empty string (new ext) \033[0m")

    for filename in li:
        if os.path.isdir(os.path.join(filename)):
            name, ext = os.path.splitext(filename)
            if ext in string.whitespace:
                FileList.append(filename)	
    if FileList == []:
        sys.exit("\033[31;1m[ERROR]: There is no file with no extension "
                 "in the path\033[0m")

    for filename in li:
        if os.path.join(filename):
            name, ext = os.path.splitext(filename)
            FileExist.append("%s.%s" %(name,newext))
    print FileExist
    for verify in FileExist:	
        if FileExist in li:
            print ("\033[31;1m[ERROR]: The object already exists with the "
                  "same name\033[0m")
            if os.path.isfile(verify):
                sys.exit('        \033[31;1m  and is a file\033[0m')
            elif os.path.isdir(verify):
                sys.exit('        \033[31;1m  and is a directory\033[0m')
                
    for filename in FileList:
        try:
            os.rename(filename, "%s.%s" % (filename, newext))
        except OSError: #'[Errno 21] Is a directory'
                sys.exit("\033[31;1m[ERROR]:  There is a file with the "
                         "same name\033[0m")
    sys.exit()
